codeowners/src/codeowners.rs

```diff
@@ -121,6 +155,58 @@ impl BindingsOwners {
     }
 }
 
+async fn associate_codeowners(
+    codeowners_matchers: Arc<HashMap<String, Option<Owners>>>,
+    bundle_upload_id_and_file_path: BundleUploadIDAndFilePath,
+) -> Vec<String> {
+    let (bundle_upload_id, file) = bundle_upload_id_and_file_path;
+    let codeowners_matcher = codeowners_matchers.get(&bundle_upload_id);
+    match (codeowners_matcher, &file) {
+        (Some(Some(owners)), Some(file)) => match owners {
+            Owners::GitHubOwners(gho) => gho
+                .of(file)
+                .unwrap_or_default()
+                .iter()
+                .map(ToString::to_string)
+                .collect(),
+            Owners::GitLabOwners(glo) => glo
+                .of(file)
+                .unwrap_or_default()
+                .iter()
+                .map(ToString::to_string)
+                .collect(),
+        },
+        _ => Vec::new(),
+    }
+}
```

Comment:
Same comment as before, I don't think that `BundleId` is related to codeowners, so we shouldn't have logic here related to it here. This method can really get simplified to the following:

```suggestion
fn associate_codeowners<T: AsRef<str>>(
    owners: &Owners,
    file: T,
) -> Vec<String> {
    match owners {
        Owners::GitHubOwners(gho) => gho
            .of(file)
            .unwrap_or_default()
            .iter()
            .map(ToString::to_string)
            .collect(),
        Owners::GitLabOwners(glo) => glo
            .of(file)
            .unwrap_or_default()
            .iter()
            .map(ToString::to_string)
            .collect(),
    }
}```